src/rust/db/src/lib.rs

```diff
@@ -19,61 +22,136 @@ pub mod schema;
 
 pub type Result<T> = std::result::Result<T, DbError>;
 
-pub fn establish_connection(url: String) -> Result<PgConnection> {
-    PgConnection::establish(&url).map_err(DbError::ConnectionError)
+pub struct EconiaDb(PgConnection);
```

Comment:
What value does `EconiaDb` add? What's the issue with just using the `PgConnection` on its own

Reply:
Then the methods for adding are implemented on the db type rather than just taking conn over and over as the first parameter. Feels cleaner.

Reply:
But if you need to use diesel, which you definitely do, you still need to access `PgConnection`, and you're forced to do `&mut *db` in place of `conn`. Overall this is just adding needless abstraction in code where you can't really abstract the database connection away.

Reply:
done i reverted this bit